homeassistant/components/hdmi_cec/switch.py

```diff
@@ -22,23 +22,25 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         add_entities(entities, True)
 
 
-class CecSwitchDevice(CecDevice, SwitchEntity):
+class CecSwitchDevice(CecEntity, SwitchEntity):
     """Representation of a HDMI device as a Switch."""
 
     def __init__(self, device, logical) -> None:
         """Initialize the HDMI device."""
-        CecDevice.__init__(self, device, logical)
+        CecEntity.__init__(self, device, logical)
         self.entity_id = f"{DOMAIN}.hdmi_{hex(self._logical_address)[2:]}"
 
     def turn_on(self, **kwargs) -> None:
         """Turn device on."""
         self._device.turn_on()
         self._state = STATE_ON
+        self.async_write_ha_state()
```

Comment:
I thought this function was only usable in an async context 

https://developers.home-assistant.io/docs/integration_fetching_data/

Reply:
It seemed to work (`async_write_ha_state()` is not a coroutine, just a callback), but based on the linked docs I shouldn't have done it that way.  I replaced these with a call `self.schedule_update_ha_state()` which also works.

Also updated `CecSwitchDevice` to `CecSwitchEntity` and `CecPlayerDevice` to `CecPlayerEntity`.